tools/CoqMakefile.in

```diff
@@ -99,11 +99,14 @@ AFTER ?=
 # FIXME this should be generated by Coq (modules already linked by Coq)
 CAMLDONTLINK=camlp5.gramlib,unix,str
 
+# XXX: This should correspond to CAMLFLAGS in configure.
+COQ_BASE_OCAML_FLAGS=-rectypes -thread -safe-string
```

Comment:
Since this is the value of CAMLFLAGS as generated by ./configure, then this variable should probably be part of Envars.print_config (https://github.com/coq/coq/blob/master/lib/envars.ml#L205) and hence be written `Makefile.conf` as someting like `COQMF_CAMLFLAGS_COQ` and renamed in this file to `CAMLFLAGS_COQ`.
If we do so, the day `CAMLFLAGS` changes value no fix to `coq_makefile` has to be made.

Reply:
I agree, but I am not 100% sure on what is available to `coq_makefile`, please feel free to push on top of this branch.

Meanwhile, we can test the hotfix.